File: c++/two-sum.cpp
#include <map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mapping;
        vector<int> res;
        int i = 0;
        for(i=0; i < nums.size(); i++){
            if(mapping.find(target - nums[i]) != mapping.end()){
                res.push_back(mapping[target - nums[i]]);
                res.push_back(i);
                return res;
            }
            else{
                mapping[nums[i]] = i;
            }
        }
        return {};
    }
};

File: python/two-sum.py
# -*- coding: utf-8 -*-
"""
Created on Tue Mar 10 10:33:17 2020

@author: shrey
"""

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        mapping = {}
        
        for i, num in enumerate(nums):
            comp_num = target - num
            if comp_num in mapping:
                return [i, mapping[comp_num]]
            else:
                mapping[num] = i
                